client/components/LearnerViewDirective.js

```diff
@@ -469,21 +465,17 @@ tie.directive('learnerView', [function() {
           loadQuestion(questionId, questionSet.getIntroductionParagraphs());
         };
 
-        var callSolutionHandlerService = function() {
-          var code = this.code;
-          SolutionHandlerService.processSolutionAsync(
-              prompts[currentPromptIndex], code,
-              question.getAuxiliaryCode(language), language
-              ).then(setFeedback);
-        };
-
         $scope.submitCode = function(code) {
-          $scope.hideDotDiv = false;
-          $timeout(
-              function(){feedbackDiv.scrollTop = feedbackDiv.scrollHeight + 17},
-              0);
-          var serviceCall = callSolutionHandlerService.bind({'code': code});
-          $timeout(serviceCall, 1000);
+          $scope.loadingIndicatorIsShown = true;
+          var additionalHeightForLoadingIndicator = 17;
+          $timeout(function(){
+              feedbackDiv.scrollTop = feedbackDiv.scrollHeight +
```

Comment:
Very minor nit: I think we're indenting by 2 throughout the codebase, not 4. (Sorry, this will be automatically handled once we get a linter in, but thought it worth mentioning so that the pattern doesn't propagate.)